tools/mason/MasonInit.chpl

```diff
@@ -37,44 +37,71 @@ Initialises a library project in a project directory
 */
 proc masonInit(args) throws {
   try! {
-    var name = '';
+    var dirName = '';
     var show = false;
+    var packageName = '';
+    var countArgs = 2;
     for arg in args[2..] {
-      if arg == '-h' || arg == '--help' {
-        masonInitHelp();
-        exit();
-      }
-      else if arg == '--show' {
-        show = true;
-      }
-      else {
-        name = arg;
+      countArgs += 1;
+      select (arg) {
+        when '-h' {
+          masonNewHelp();
+          exit();
+        }
+        when '--help' {
+          masonNewHelp();
+          exit();
+        }
+        when '--show' {
+          show = true;
+        }
+        when '--name' {
+          packageName = args[countArgs];
+          dirName = args[2];
+        }
+        otherwise {
+          if arg.startsWith('--name') {
+            var res = arg.split("=");
+            packageName = res[2];
+            dirName = args[2];
+          }
+          if packageName.length == 0 then {
+            dirName = arg;
+            packageName = dirName;
+          }
+        }
       }
     }
-    if name == '' {
+
+    if dirName.startsWith('--name') then dirName = '';
```

Comment:
Does this ever actually happen? I would expect it not to.

Reply:
Without this condition, 
```
mkdir project-Src
cd project-Src
mason init --name=project      # packageName = project , dirName = --name=project
ERROR !
```
With this condition, we use packageName to initialise the module

Reply:
shouldn't  the case above catch this and set `dir` appropriately though? 

` if arg.startsWith('--name=') { ... }`

https://github.com/chapel-lang/chapel/pull/14997/files#diff-ef69f825f0dea9e83ad07d0e0f9ae4acR63